genet/core.py

```diff
@@ -2207,38 +2207,42 @@ def validation_report_for_node_elevation(self, low_limit=-50, mont_blanc_height=
         :param mont_blanc_height: defaults to 4809m is the height of Mont Blank, can optionally set a different value
-        :return: dict
+        :return: dict, with 2 data subsets - summary statistics, and extreme values lists
         """
         graph = self.subgraph_on_link_conditions(conditions={'modes': all}, mixed_dtypes=True)
         gdfs = geojson.generate_geodataframes(graph)
         nodes = gdfs['nodes']
 
-        elevation_list = nodes['z'].to_list()
-        min_value = np.min(elevation_list)
-        max_value = np.max(elevation_list)
-        mean = np.mean(elevation_list)
-        median = np.median(elevation_list)
-
-        nodes_dictionary = dict(zip(nodes['id'], nodes['z']))
-        too_high = {}
-        too_low = {}
-
-        for node_id, node_elev in nodes_dictionary.items():
-            if node_elev < low_limit:
-                too_low[node_id] = node_elev
-            elif node_elev > mont_blanc_height:
-                too_high[node_id] = node_elev
-
-        report = {
-            'summary': {'total_nodes': len(elevation_list),
-                        'min_value': int(min_value),
-                        'max_value': int(max_value),
-                        'mean': int(mean),
-                        'median': int(median),
-                        'extremely_high_values_count': len(too_high),
-                        'extremely_low_values_count': len(too_low)},
-
-            'values': {'extremely_high_values_dict': too_high,
-                       'extremely_low_values_dict': too_low}}
+        if 'z' in list(nodes.columns):
+            elevation_list = nodes['z'].to_list()
+            min_value = np.min(elevation_list)
+            max_value = np.max(elevation_list)
+            mean = np.mean(elevation_list)
+            median = np.median(elevation_list)
+
+            nodes_dictionary = dict(zip(nodes['id'], nodes['z']))
+            too_high = {}
+            too_low = {}
+
+            for node_id, node_elev in nodes_dictionary.items():
+                if node_elev < low_limit:
+                    too_low[node_id] = node_elev
+                elif node_elev > mont_blanc_height:
+                    too_high[node_id] = node_elev
+
+            report = {
+                'summary': {'total_nodes': len(elevation_list),
+                            'min_value': int(min_value),
+                            'max_value': int(max_value),
+                            'mean': int(mean),
+                            'median': int(median),
+                            'extremely_high_values_count': len(too_high),
+                            'extremely_low_values_count': len(too_low)},
+
+                'values': {'extremely_high_values_dict': too_high,
+                           'extremely_low_values_dict': too_low}}
+
+            return report
 
-        return report
+        else:
+            logging.info('Network nodes do not contain elevation data. Cannot generate validation report.')
```

Comment:
@mfitz I have a logging message here if there is no elevation data in the network?